lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs

```diff
@@ -133,17 +133,22 @@ Checkpoint
     checkpointBlockHeight       Word32       sql=block_height
     checkpointGenesisHash       BlockId      sql=genesis_hash
     checkpointGenesisStart      UTCTime      sql=genesis_start
-    checkpointFeePolicy         W.FeePolicy  sql=fee_policy
     checkpointSlotLength        Word64       sql=slot_length
     checkpointEpochLength       Word32       sql=epoch_length
-    checkpointTxMaxSize         Word16       sql=tx_max_size
     checkpointEpochStability    Word32       sql=epoch_stability
     checkpointActiveSlotCoeff   Double       sql=active_slot_coeff
 
     Primary checkpointWalletId checkpointSlot
     Foreign Wallet checkpoint checkpointWalletId ! ON DELETE CASCADE
     deriving Show Generic
 
+TxParameters
+    txParametersWalletId          W.WalletId  sql=wallet_id
+    txParametersFeePolicy         W.FeePolicy sql=fee_policy
+    txParametersTxMaxSize         Word16      sql=tx_max_size
+    Primary txParametersWalletId
+    deriving Show Generic
```

Comment:
There's no slot or epoch reference here. How do we deal with rollbacks, especially on epoch boundaries?
This is the reason why parameters ended up being stored with checkpoints in the first place (although a bit excessive since most of the stored parameters are actually static ... ). But these needs to be consistent with checkpoints, so when we rollback, we ideally need to also rollback to the previous tx parameters values. 

So we either need to store them with checkpoints, or make sure to keep this table and the checkpoint one in sync :thinking: ... Tx parameters however don't necessarily change on every rollback though, but only when we cross an epoch boundary.

Reply:
Also, on a slightly different note, if we really end up with a different table here, what about choosing a slightly more general name "ProtocolParameters" or "UpdatableParameters" because, there could be more in practice and it'll be sad to have the table called "TxParameters" if we end up adding more params that aren't tx-specific. 

My suggestion would be however to do the opposite here:

a) Keep the tx-specific parameters inside the `checkpoint` table. 
b) Move the static parameters (slot length, epoch length...) inside a "static" table (or just keep them in a record in-memory; we know mainnet, and testnet are provided at runtime anyway).

Reply:
Do we actually need to store the `TxParameters` in the DB?

Why not refetch them on each tx creation (or somewhat cached), and not have to worry about rollbacks and consistency?

Reply:
@KtorZ response to your first comment is below.

@Anviking You are right there is no need to store these parameters in the DB. However, if we do cache them in the DB, it is possible to create transactions without the network layer being active.